### ISSUE TITLE
[Bug]: Button styling adjustments (missing selected state, icon color, icon style)

### ISSUE THREAD

**Issue by redongreen** [microsoft/fluentui#25032]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
https://fluentuipr.z22.web.core.windows.net/heads/master/public-docsite-v9/storybook/index.html?path=/docs/components-button-button--default#icon
```

### Are you reporting Accessibility issue?

no

### Reproduction

https://fluentuipr.z22.web.core.windows.net/heads/master/public-docsite-v9/storybook/index.html?path=/docs/components-button-button--default

### 1. Primary, secondary/default button variant icon change

**Current:** While hovered the icon changes from outlined -> filled. [See hover behavior in codesandbox](https://codesandbox.io/s/cx4hx8?file=/example.tsx).
<img width="293" alt="image" src="https://user-images.githubusercontent.com/20573265/193288606-0615a2e2-743a-48d0-af31-a7a2463df614.png">

**Expected:** The primary and default/secondary button variant should not change from outlined to filled icon in hover, pressed, selected state. 
<img width="561" alt="image" src="https://user-images.githubusercontent.com/20573265/193288928-d00f556f-c645-4a28-bf36-e8761043d1e7.png">

___
### 2. Subtle variant icon color in hover, pressed, and selected states

**Current:** While in hovered, pressed, and selected state the icon is styled using neutral color.
<img width="220" alt="image" src="https://user-images.githubusercontent.com/20573265/193293373-8ac496c6-6055-43ee-8780-9c6a5ce26c1f.png">

**Expected:** While in hovered, pressed, and selected state the icon should be styled using brand accent color according to our design specs. [Link to Subtle button spec](https://www.figma.com/file/Nj9EBBvOZmS11zKNJfilVR/Button?node-id=4637%3A19935).
<img width="432" alt="image" src="https://user-images.githubusercontent.com/20573265/193294001-5f4aa307-b914-43af-81b6-05736a4dbbbd.png">

___
### 3. Selected state missing from all menu button variants

**Current:** While the button is selected the styling of the button defaults to the rest state. The example below shows the default/secondary variant button, but this also applies to primary, subtle, outline, and transparent buttons.
<img width="393" alt="image" src="https://user-images.githubusercontent.com/20573265/193290446-5da33801-aea5-40b0-8e81-f4f8fd4ccdcf.png">

**Expected:** While selected, the button should follow the styling provided by our specs. [Link to spec, see token allocation under "Toggle press on" section](https://www.figma.com/file/Nj9EBBvOZmS11zKNJfilVR/Button?node-id=4637%3A17423).
<img width="261" alt="image" src="https://user-images.githubusercontent.com/20573265/193291430-cf805508-0b4a-44a4-b240-2ecb816edcac.png">

### Logs

```shell
https://fluentuipr.z22.web.core.windows.net/heads/master/public-docsite-v9/storybook/index.html?path=/docs/components-button-button--default
```

### Requested priority

Normal

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by kkakroo** [comment]
@khmakoto , can i have a look into this

**Comment by khmakoto** [comment]
@kkakroo feel free to take a shot at fixing this!

**Comment by kkakroo** [comment]
Thanks @khmakoto.

For the 1st part of the bug, it seems to be happening as we are using `bundleIcon` in the story and the button has specific styling for handling those scenarios added in [this](https://github.com/microsoft/fluentui/pull/22587/files#diff-f6fb812b1e0ffe16b643e1d8f8db1240a2d67e6d7492df459a4597cac9b5aca5) PR. 

So not using the `bundleIcon` icon and using the normal icon as is does not have those hover effects.
@khmakoto, need your thoughts on this for the approach to take in this scenario.

For the 2nd and 3rd parts of the bug, I do not have access to the Figma file (provided in the link) which might give an idea of the token it needs to be changed to.

**Comment by khmakoto** [comment]
Hi @kkakroo, I'm actually fixing the first part of the bug as part of #25479.

You're right about the second and third parts needing the Figma. Unfortunately, they are not public, so I think I'll need to work on this myself. Thank you very much for willing to contribute though!

**Comment by khmakoto** [comment]
Issues 1 and 2 are fixed by #25479. I'll create a separate PR to fix issue 3.
